### ISSUE TITLE
Workspace Mode - Requires Data Refresh

### ISSUE THREAD

**Issue by MaddyTP** [TabularEditor/TabularEditor3#697]
When connecting to a workspace in Workspace Mode it clears the data in PowerBI service requiring a refresh and breaking existing reports which rely on the data model.

**Comment by otykier** [comment]
Hi Matt,
I’m afraid there’s been a misunderstanding. Workspace Mode is not intended to be used with a production dataset. Rather, it is a way for each model developer to have a personal dataset attached to TE3, where they can quickly make and test changes. In general, you should not touch any datasets that have production reports attached to them, with Tabular Editor - unless you’re performing a deployment. More information here: https://docs.tabulareditor.com/te3/tutorials/workspace-mode.html

TE3 does not explicitly clear data in a workspace database when the file is loaded, but it does perform a deployment of the model metadata from the file that was loaded, unto the chosen workspace dataset. As such, depending on the differences between the loaded metadata and the existing dataset, tables or partitions may be dropped, thus giving the impression that data has been cleared.

**Comment by MaddyTP** [comment]
Hmm...interesting perspective.  We are using workspace mode on our development model so we can make quick changes and verify in report building.  Models are then progressed through a PowerBI pipeline for production.  

Workspace mode is fantastic for ensuring changes are saved in both places.  It would be nice if it didn't "corrupt" the data model when connecting.

**Comment by otykier** [comment]
Have you tried performing a simple Calculate Refresh after connecting? This should ensure that any calculated columns, calculated tables or modified relationships will be updated to a queryable state. But as long as each developer uses their own workspace dataset(s), this shouldn’t be an issue.

**Comment by MaddyTP** [comment]
Yes, calculating the model does fix the issue.  It would be nice if the model recalculated automatically after synchronizing so the manual calculate wasn't required.  As it stands, without manually recalculating the model it is broken upon synchronization.

**Comment by otykier** [comment]
We may consider that as an option, since a calculate refresh can take a long time to complete, depending on the size and complexity of the calculations involved. Not something you’d want to happen every time you load a model in all cases.
